### ISSUE TITLE
`yarn start:contract` fails: Proposal rejected at propose-start-contract.sh

### ISSUE THREAD

**Issue by migueldingli1997** [Agoric/dapp-offer-up#44]
On some occasions `yarn start:contract` fails at the last moment with a proposal rejection.

I've also noticed 'account sequence mismatch' errors occurring on the node logs, indicating the possibility that we're using a single account to sign more than one transaction in a single block. Perhaps the voting for the proposal is failing, causing the proposal to get rejected. But this might also not be related to the problems I'm getting.

Usually, just running `yarn start:contract` again resolves the issue and the contract gets deployed, but sometimes it doesn't, causing an annoying setback.

<details>
<summary>Attaching logs from start:contract... (click me)</summary>

```bash
...
scripts/propose-start-contract.sh
+ cd /workspace/contract
+ SCRIPT=start-game1.js
+ PERMIT=start-game1-permit.json
+ ls -sh start-game1.js start-game1-permit.json
4.0K start-game1-permit.json  8.0K start-game1.js
++ agd query gov proposals --output json
++ jq -c '.proposals | length | .+1'
+ PROPOSAL=9
+ make fund-acct
make[1]: Entering directory '/workspace/contract'
agd tx bank send validator agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5 321000000ubld \
  --keyring-backend=test --chain-id=agoriclocal --gas=auto --gas-adjustment=1.2 --yes -b block \
  -o json >,tx.json
gas estimate: 76758
jq '{code: .code, height: .height}' ,tx.json
{
  "code": 0,
  "height": "1138"
}
make[1]: Leaving directory '/workspace/contract'
+ agd tx gov submit-proposal swingset-core-eval start-game1-permit.json start-game1.js '--title=Start Game Place Contract' '--description=Evaluate start-game1.js' --deposit=10000000ubld --gas=auto --gas-adjustment=1.2 --from user1 --chain-id agoriclocal --keyring-backend=test --yes -b block
gas estimate: 1004055
code: 0
codespace: ""
data: 0A2B0A252F636F736D6F732E676F762E763162657461312E4D73675375626D697450726F706F73616C12020809
events:
- attributes:
  - index: true
    key: ZmVl
    value: ""
  type: tx
- attributes:
  - index: true
    key: YWNjX3NlcQ==
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1LzEz
  type: tx
- attributes:
  - index: true
    key: c2lnbmF0dXJl
    value: Z3BJN1dEVHRvZGZxcVp5ZVYzSmVzYkMyZUF1UTRQeW5BZHRHUkNwZmljeEVocmNDVWhNbWRXd0d2V3p4UENnQzk4Mmw3Q3BhbmhJUGNsSHVYenJTcFE9PQ==
  type: tx
- attributes:
  - index: true
    key: YWN0aW9u
    value: L2Nvc21vcy5nb3YudjFiZXRhMS5Nc2dTdWJtaXRQcm9wb3NhbA==
  type: message
- attributes:
  - index: true
    key: cHJvcG9zYWxfaWQ=
    value: OQ==
  type: submit_proposal
- attributes:
  - index: true
    key: c3BlbmRlcg==
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: coin_spent
- attributes:
  - index: true
    key: cmVjZWl2ZXI=
    value: YWdvcmljMTBkMDd5MjY1Z21tdXZ0NHowdzlhdzg4MGpuc3I3MDBqZ2wzNng5
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: coin_received
- attributes:
  - index: true
    key: cmVjaXBpZW50
    value: YWdvcmljMTBkMDd5MjY1Z21tdXZ0NHowdzlhdzg4MGpuc3I3MDBqZ2wzNng5
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: transfer
- attributes:
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  type: message
- attributes:
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  - index: true
    key: cHJvcG9zYWxfaWQ=
    value: OQ==
  type: proposal_deposit
- attributes:
  - index: true
    key: bW9kdWxl
    value: Z292ZXJuYW5jZQ==
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  type: message
- attributes:
  - index: true
    key: cHJvcG9zYWxfdHlwZQ==
    value: Q29yZUV2YWw=
  - index: true
    key: dm90aW5nX3BlcmlvZF9zdGFydA==
    value: OQ==
  type: submit_proposal
gas_used: "833693"
gas_wanted: "1004055"
height: "1140"
info: ""
logs:
- events:
  - attributes:
    - key: receiver
      value: agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9
    - key: amount
      value: 10000000ubld
    type: coin_received
  - attributes:
    - key: spender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: amount
      value: 10000000ubld
    type: coin_spent
  - attributes:
    - key: action
      value: /cosmos.gov.v1beta1.MsgSubmitProposal
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: module
      value: governance
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    type: message
  - attributes:
    - key: amount
      value: 10000000ubld
    - key: proposal_id
      value: "9"
    type: proposal_deposit
  - attributes:
    - key: proposal_id
      value: "9"
    - key: proposal_type
      value: CoreEval
    - key: voting_period_start
      value: "9"
    type: submit_proposal
  - attributes:
    - key: recipient
      value: agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: amount
      value: 10000000ubld
    type: transfer
  log: ""
  msg_index: 0
raw_log: '[{"events":[{"type":"coin_received","attributes":[{"key":"receiver","value":"agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9"},{"key":"amount","value":"10000000ubld"}]},{"type":"coin_spent","attributes":[{"key":"spender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"amount","value":"10000000ubld"}]},{"type":"message","attributes":[{"key":"action","value":"/cosmos.gov.v1beta1.MsgSubmitProposal"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"module","value":"governance"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"}]},{"type":"proposal_deposit","attributes":[{"key":"amount","value":"10000000ubld"},{"key":"proposal_id","value":"9"}]},{"type":"submit_proposal","attributes":[{"key":"proposal_id","value":"9"},{"key":"proposal_type","value":"CoreEval"},{"key":"voting_period_start","value":"9"}]},{"type":"transfer","attributes":[{"key":"recipient","value":"agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"amount","value":"10000000ubld"}]}]}]'
timestamp: ""
tx: null
txhash: E85886D45D1CF382513BF19D7B8FB7E2888C801AB65689B0AEEC066CCC6179F0
+ set +x
ENV_SETUP starting
ENV_SETUP finished
waiting for block...
1
block produced
done
waiting for block...
1
block produced
done
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: B9D19A5A17010834891DB965155A70460C2CFBE63F15689C93A9D952C29244F7
waiting for block...
1
block produced
done
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: F65FBDE3F81D83A01AC633FB7D03ABAAF7D2D58189BD37D439AEAB71EEBA7A80
waiting for block...
1
block produced
done
Proposal rejected
make: *** [Makefile:75: start-contract] Error 1
make: Leaving directory '/workspace/contract'
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
</details>

<details>
<summary>Attaching logs from Docker container at the same block range... (click me)</summary>

```bash
...
2024-01-12 13:14:03 2024-01-12T12:14:03.421Z block-manager: block 1135 begin
2024-01-12 13:14:03 2024-01-12T12:14:03.461Z block-manager: block 1135 commit
2024-01-12 13:14:04 2024-01-12T12:14:04.439Z block-manager: block 1136 begin
2024-01-12 13:14:04 2024-01-12T12:14:04.479Z block-manager: block 1136 commit
2024-01-12 13:14:05 2024-01-12T12:14:05.454Z block-manager: block 1137 begin
2024-01-12 13:14:05 2024-01-12T12:14:05.491Z block-manager: block 1137 commit
2024-01-12 13:14:06 2024-01-12T12:14:06.469Z block-manager: block 1138 begin
2024-01-12 13:14:06 2024-01-12T12:14:06.514Z block-manager: block 1138 commit
2024-01-12 13:14:07 2024-01-12T12:14:07.487Z block-manager: block 1139 begin
2024-01-12 13:14:07 2024-01-12T12:14:07.526Z block-manager: block 1139 commit
2024-01-12 13:14:08 Approving proposals: 
2024-01-12 13:14:08 2024-01-12T12:14:08.503Z block-manager: block 1140 begin
2024-01-12 13:14:08 2024-01-12T12:14:08.719Z block-manager: block 1140 commit
2024-01-12 13:13:46 printf: usage: printf [-v var] format [arguments]
2024-01-12 13:13:57 printf: usage: printf [-v var] format [arguments]
2024-01-12 13:14:08 printf: usage: printf [-v var] format [arguments]
2024-01-12 13:14:19 printf: usage: printf [-v var] format [arguments]
2024-01-12 13:14:30 printf: usage: printf [-v var] format [arguments]
2024-01-12 13:14:09 2024-01-12T12:14:09.518Z block-manager: block 1141 begin
2024-01-12 13:14:09 2024-01-12T12:14:09.556Z block-manager: block 1141 commit
2024-01-12 13:14:10 2024-01-12T12:14:10.536Z block-manager: block 1142 begin
2024-01-12 13:14:10 2024-01-12T12:14:10.575Z block-manager: block 1142 commit
2024-01-12 13:14:11 2024-01-12T12:14:11.550Z block-manager: block 1143 begin
2024-01-12 13:14:11 2024-01-12T12:14:11.590Z block-manager: block 1143 commit
2024-01-12 13:14:12 2024-01-12T12:14:12.567Z block-manager: block 1144 begin
2024-01-12 13:14:12 2024-01-12T12:14:12.604Z block-manager: block 1144 commit
2024-01-12 13:14:13 2024-01-12T12:14:13.593Z block-manager: block 1145 begin
2024-01-12 13:14:13 2024-01-12T12:14:13.634Z block-manager: block 1145 commit
2024-01-12 13:14:14 2024-01-12T12:14:14.608Z block-manager: block 1146 begin
2024-01-12 13:14:14 2024-01-12T12:14:14.647Z block-manager: block 1146 commit
2024-01-12 13:14:15 2024-01-12T12:14:15.647Z block-manager: block 1147 begin
2024-01-12 13:14:15 2024-01-12T12:14:15.687Z block-manager: block 1147 commit
2024-01-12 13:14:16 2024-01-12T12:14:16.654Z block-manager: block 1148 begin
2024-01-12 13:14:16 2024-01-12T12:14:16.693Z block-manager: block 1148 commit
2024-01-12 13:14:17 2024-01-12T12:14:17.672Z block-manager: block 1149 begin
2024-01-12 13:14:17 2024-01-12T12:14:17.717Z block-manager: block 1149 commit
2024-01-12 13:14:18 2024-01-12T12:14:18.686Z block-manager: block 1150 begin
2024-01-12 13:14:18 2024-01-12T12:14:18.732Z block-manager: block 1150 commit
2024-01-12 13:14:19 Approving proposals: 
2024-01-12 13:14:19 2024-01-12T12:14:19.785Z block-manager: block 1151 begin
2024-01-12 13:14:19 2024-01-12T12:14:19.906Z block-manager: block 1151 commit
2024-01-12 13:14:20 2024-01-12T12:14:20.758Z block-manager: block 1152 begin
2024-01-12 13:14:20 2024-01-12T12:14:20.797Z block-manager: block 1152 commit
2024-01-12 13:14:21 2024-01-12T12:14:21.777Z block-manager: block 1153 begin
2024-01-12 13:14:21 2024-01-12T12:14:21.819Z block-manager: block 1153 commit
2024-01-12 13:14:22 2024-01-12T12:14:22.792Z block-manager: block 1154 begin
2024-01-12 13:14:22 2024-01-12T12:14:22.831Z block-manager: block 1154 commit
2024-01-12 13:14:23 2024-01-12T12:14:23.828Z block-manager: block 1155 begin
2024-01-12 13:14:23 2024-01-12T12:14:23.869Z block-manager: block 1155 commit
2024-01-12 13:14:24 2024-01-12T12:14:24.863Z block-manager: block 1156 begin
2024-01-12 13:14:24 2024-01-12T12:14:24.909Z block-manager: block 1156 commit
2024-01-12 13:14:25 2024-01-12T12:14:25.909Z block-manager: block 1157 begin
2024-01-12 13:14:25 2024-01-12T12:14:25.949Z block-manager: block 1157 commit
2024-01-12 13:14:26 2024-01-12T12:14:26.929Z block-manager: block 1158 begin
2024-01-12 13:14:26 2024-01-12T12:14:26.967Z block-manager: block 1158 commit
2024-01-12 13:14:27 2024-01-12T12:14:27.956Z block-manager: block 1159 begin
2024-01-12 13:14:27 2024-01-12T12:14:27.994Z block-manager: block 1159 commit
2024-01-12 13:14:28 2024-01-12T12:14:28.969Z block-manager: block 1160 begin
2024-01-12 13:14:29 2024-01-12T12:14:29.008Z block-manager: block 1160 commit
2024-01-12 13:14:29 2024-01-12T12:14:29.984Z block-manager: block 1161 begin
2024-01-12 13:14:30 2024-01-12T12:14:30.022Z block-manager: block 1161 commit
2024-01-12 13:14:30 Approving proposals: 
2024-01-12 13:14:31 2024-01-12T12:14:31.014Z block-manager: block 1162 begin
2024-01-12 13:14:31 2024-01-12T12:14:31.050Z block-manager: block 1162 commit
2024-01-12 13:14:32 2024-01-12T12:14:32.051Z block-manager: block 1163 begin
2024-01-12 13:14:32 2024-01-12T12:14:32.088Z block-manager: block 1163 commit
2024-01-12 13:14:33 2024-01-12T12:14:33.070Z block-manager: block 1164 begin
2024-01-12 13:14:33 2024-01-12T12:14:33.107Z block-manager: block 1164 commit
2024-01-12 13:14:34 2024-01-12T12:14:34.097Z block-manager: block 1165 begin
2024-01-12 13:14:34 2024-01-12T12:14:34.135Z block-manager: block 1165 commit
2024-01-12 13:14:35 2024-01-12T12:14:35.121Z block-manager: block 1166 begin
2024-01-12 13:14:35 2024-01-12T12:14:35.160Z block-manager: block 1166 commit
2024-01-12 13:14:36 2024-01-12T12:14:36.142Z block-manager: block 1167 begin
2024-01-12 13:14:36 2024-01-12T12:14:36.179Z block-manager: block 1167 commit
2024-01-12 13:14:37 2024-01-12T12:14:37.170Z block-manager: block 1168 begin
2024-01-12 13:14:37 2024-01-12T12:14:37.209Z block-manager: block 1168 commit
2024-01-12 13:14:38 2024-01-12T12:14:38.192Z block-manager: block 1169 begin
2024-01-12 13:14:38 2024-01-12T12:14:38.236Z block-manager: block 1169 commit
2024-01-12 13:14:39 2024-01-12T12:14:39.223Z block-manager: block 1170 begin
2024-01-12 13:14:39 2024-01-12T12:14:39.263Z block-manager: block 1170 commit
2024-01-12 13:14:40 2024-01-12T12:14:40.237Z block-manager: block 1171 begin
2024-01-12 13:14:40 2024-01-12T12:14:40.280Z block-manager: block 1171 commit
2024-01-12 13:14:41 Approving proposals: 
2024-01-12 13:14:41 2024-01-12T12:14:41.256Z block-manager: block 1172 begin
2024-01-12 13:14:41 2024-01-12T12:14:41.295Z block-manager: block 1172 commit
2024-01-12 13:14:42 2024-01-12T12:14:42.317Z block-manager: block 1173 begin
2024-01-12 13:14:42 2024-01-12T12:14:42.358Z block-manager: block 1173 commit
2024-01-12 13:14:43 2024-01-12T12:14:43.323Z block-manager: block 1174 begin
2024-01-12 13:14:43 2024-01-12T12:14:43.366Z block-manager: block 1174 commit
2024-01-12 13:14:44 2024-01-12T12:14:44.339Z block-manager: block 1175 begin
2024-01-12 13:14:44 2024-01-12T12:14:44.379Z block-manager: block 1175 commit
2024-01-12 13:14:45 2024-01-12T12:14:45.357Z block-manager: block 1176 begin
2024-01-12 13:14:45 2024-01-12T12:14:45.405Z block-manager: block 1176 commit
2024-01-12 13:14:46 2024-01-12T12:14:46.373Z block-manager: block 1177 begin
2024-01-12 13:14:46 2024-01-12T12:14:46.416Z block-manager: block 1177 commit
2024-01-12 13:14:47 2024-01-12T12:14:47.389Z block-manager: block 1178 begin
2024-01-12 13:14:47 2024-01-12T12:14:47.432Z block-manager: block 1178 commit
2024-01-12 13:14:48 2024-01-12T12:14:48.405Z block-manager: block 1179 begin
2024-01-12 13:14:48 2024-01-12T12:14:48.449Z block-manager: block 1179 commit
2024-01-12 13:14:49 2024-01-12T12:14:49.422Z block-manager: block 1180 begin
2024-01-12 13:14:49 2024-01-12T12:14:49.467Z block-manager: block 1180 commit
2024-01-12 13:14:50 2024-01-12T12:14:50.440Z block-manager: block 1181 begin
2024-01-12 13:14:50 2024-01-12T12:14:50.484Z block-manager: block 1181 commit
2024-01-12 13:14:51 2024-01-12T12:14:51.457Z block-manager: block 1182 begin
2024-01-12 13:14:51 2024-01-12T12:14:51.505Z block-manager: block 1182 commit
2024-01-12 13:14:52 Approving proposals: 
2024-01-12 13:14:52 2024-01-12T12:14:52.477Z block-manager: block 1183 begin
2024-01-12 13:14:52 2024-01-12T12:14:52.522Z block-manager: block 1183 commit
2024-01-12 13:14:53 2024-01-12T12:14:53.493Z block-manager: block 1184 begin
2024-01-12 13:14:53 2024-01-12T12:14:53.537Z block-manager: block 1184 commit
2024-01-12 13:14:54 2024-01-12T12:14:54.511Z block-manager: block 1185 begin
2024-01-12 13:14:54 2024-01-12T12:14:54.557Z block-manager: block 1185 commit
...
```
</details>

**Comment by dckc** [comment]
Thanks for reporting your experience.

Please stay tuned for follow-up.

cc @aj-agoric @sufyaankhan

**Comment by gacogo** [comment]
Does this solve your issue @migueldingli1997  https://github.com/Agoric/dapp-offer-up/issues/43#issuecomment-1902573075

**Comment by 0xpatrickdev** [comment]
A potential solution here - or something that should be done regardless - is breaking down `yarn start:contract` (`yarn docker:make clean start-contract print-key`) into separate pieces. 

This happens to be intermittently as well, and a pain to have to start from square 0. The distinctly separate steps seem to be - 
1. build the proposal / bundles (`./scripts/build-proposal.sh`)
2. install the bundles
3. submit CoreEval
4. deposit and vote

Often times, it seems to get tripped up on voting (out of gas, inactive proposal). It would be helpful if I can retry from step 3, without needing to mint more IST or reinstall existing bundles.

Other thoughts - 
- `voteLatestProposalAndWait` takes care of depositing (and will attempt a deposit even if the proposal is active), so maybe the CoreEval should not include a deposit. This might eliminate instances of 'inactive proposal'. 
- `yarn start:docker` takes a bit of time to get going - maybe `yarn start:contract` should do a check that a certain block height has been reached before starting step 2.

**Comment by dckc** [comment]
[...]
> 1. build the proposal / bundles (`./scripts/build-proposal.sh`)
> 2. install the bundles
> 3. submit CoreEval
> 4. deposit and vote
> 
> Often times, it seems to get tripped up on voting (out of gas, inactive proposal).

Here's hoping we can have a separate issue for each root cause we identify.
IOU an issue for insufficient IST when installing bundles.

**Comment by dckc** [comment]
> It would be helpful if I can retry from step 3, without needing to mint more IST or reinstall existing bundles.

The `Makefile` could/should scribble into the filesystem after each step to mark progress.

I make some progress in that direction in ag-power-tools:
 - [feat(launchIt): deploy bundle using Makefile rule](https://github.com/agoric-labs/ag-power-tools/pull/7/commits/51d3cd660d98f54466a9a2ba51e2575f4171cfe4)

It writes to `xyz.json.installed` whenever it installs the `xyz.json` bundle.

**Comment by dckc** [comment]
> * `yarn start:docker` takes a bit of time to get going - maybe `yarn start:contract` should do a check that a certain block height has been reached before starting step 2.

See:

 - #27

**Comment by dckc** [comment]
> I've also noticed 'account sequence mismatch' errors occurring on the node logs

The logs above don't include that string.

I don't see why the proposal failed... unless the proposal timed out?

Does anybody else have thoughts on root cause here?

**Comment by migueldingli1997** [comment]
> > I've also noticed 'account sequence mismatch' errors occurring on the node logs
> 
> The logs above don't include that string.

Hey @dckc it's hard to know if the account sequence mismatch errors are at all related to the proposal rejection.

I got an example of an account sequence mismatch happening error right before `yarn start:contract` concludes. In this case it wasn't on the node logs, and it did not affect the proposal and the contract started successfully. If I run into such an error again on the node side I'll paste it here.

<details>
<summary>(click me)</summary>

```bash
...
+ agd tx gov submit-proposal swingset-core-eval start-game1-permit.json start-game1.js '--title=Start Game Place Contract' '--description=Evaluate start-game1.js' --deposit=10000000ubld --gas=auto --gas-adjustment=1.2 --from user1 --chain-id agoriclocal --keyring-backend=test --yes -b block
gas estimate: 1014733
code: 0
codespace: ""
data: 0A2B0A252F636F736D6F732E676F762E763162657461312E4D73675375626D697450726F706F73616C12020809
events:
- attributes:
  - index: true
    key: ZmVl
    value: ""
  type: tx
- attributes:
  - index: true
    key: YWNjX3NlcQ==
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1LzEz
  type: tx
- attributes:
  - index: true
    key: c2lnbmF0dXJl
    value: bjdxejdUWGhQdXQwRmgrUVBYNEs5ejNaOFpDdjgxcUpzTVRuQklwTG9xY3BBcEgwdWl1bTc4Zkl2MmxPRy84QnlsNzJXS21rd3RYUHFYV0JEYktVYmc9PQ==
  type: tx
- attributes:
  - index: true
    key: YWN0aW9u
    value: L2Nvc21vcy5nb3YudjFiZXRhMS5Nc2dTdWJtaXRQcm9wb3NhbA==
  type: message
- attributes:
  - index: true
    key: cHJvcG9zYWxfaWQ=
    value: OQ==
  type: submit_proposal
- attributes:
  - index: true
    key: c3BlbmRlcg==
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: coin_spent
- attributes:
  - index: true
    key: cmVjZWl2ZXI=
    value: YWdvcmljMTBkMDd5MjY1Z21tdXZ0NHowdzlhdzg4MGpuc3I3MDBqZ2wzNng5
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: coin_received
- attributes:
  - index: true
    key: cmVjaXBpZW50
    value: YWdvcmljMTBkMDd5MjY1Z21tdXZ0NHowdzlhdzg4MGpuc3I3MDBqZ2wzNng5
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  type: transfer
- attributes:
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  type: message
- attributes:
  - index: true
    key: YW1vdW50
    value: MTAwMDAwMDB1Ymxk
  - index: true
    key: cHJvcG9zYWxfaWQ=
    value: OQ==
  type: proposal_deposit
- attributes:
  - index: true
    key: bW9kdWxl
    value: Z292ZXJuYW5jZQ==
  - index: true
    key: c2VuZGVy
    value: YWdvcmljMXhlMjY5eTNmaHllOG5ybGR1ZjgyNndnbjQ5OXk2d21udjMydHc1
  type: message
- attributes:
  - index: true
    key: cHJvcG9zYWxfdHlwZQ==
    value: Q29yZUV2YWw=
  - index: true
    key: dm90aW5nX3BlcmlvZF9zdGFydA==
    value: OQ==
  type: submit_proposal
gas_used: "842591"
gas_wanted: "1014733"
height: "1110"
info: ""
logs:
- events:
  - attributes:
    - key: receiver
      value: agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9
    - key: amount
      value: 10000000ubld
    type: coin_received
  - attributes:
    - key: spender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: amount
      value: 10000000ubld
    type: coin_spent
  - attributes:
    - key: action
      value: /cosmos.gov.v1beta1.MsgSubmitProposal
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: module
      value: governance
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    type: message
  - attributes:
    - key: amount
      value: 10000000ubld
    - key: proposal_id
      value: "9"
    type: proposal_deposit
  - attributes:
    - key: proposal_id
      value: "9"
    - key: proposal_type
      value: CoreEval
    - key: voting_period_start
      value: "9"
    type: submit_proposal
  - attributes:
    - key: recipient
      value: agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9
    - key: sender
      value: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5
    - key: amount
      value: 10000000ubld
    type: transfer
  log: ""
  msg_index: 0
raw_log: '[{"events":[{"type":"coin_received","attributes":[{"key":"receiver","value":"agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9"},{"key":"amount","value":"10000000ubld"}]},{"type":"coin_spent","attributes":[{"key":"spender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"amount","value":"10000000ubld"}]},{"type":"message","attributes":[{"key":"action","value":"/cosmos.gov.v1beta1.MsgSubmitProposal"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"module","value":"governance"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"}]},{"type":"proposal_deposit","attributes":[{"key":"amount","value":"10000000ubld"},{"key":"proposal_id","value":"9"}]},{"type":"submit_proposal","attributes":[{"key":"proposal_id","value":"9"},{"key":"proposal_type","value":"CoreEval"},{"key":"voting_period_start","value":"9"}]},{"type":"transfer","attributes":[{"key":"recipient","value":"agoric10d07y265gmmuvt4z0w9aw880jnsr700jgl36x9"},{"key":"sender","value":"agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5"},{"key":"amount","value":"10000000ubld"}]}]}]'
timestamp: ""
tx: null
txhash: BE6B301E67489DF441428C3DA461E0C1F2F12CF87A3D30848DFD9B82A95416B4
+ set +x
ENV_SETUP starting
ENV_SETUP finished
waiting for block...
1
block produced
done
waiting for block...
1
block produced
done
code: 32
codespace: sdk
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: 'account sequence mismatch, expected 41, got 40: incorrect account sequence'
timestamp: ""
tx: null
txhash: B9D19A5A17010834891DB965155A70460C2CFBE63F15689C93A9D952C29244F7
waiting for block...
1
block produced
done
code: 0
codespace: ""
data: ""
events: []
gas_used: "0"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: '[]'
timestamp: ""
tx: null
txhash: F65FBDE3F81D83A01AC633FB7D03ABAAF7D2D58189BD37D439AEAB71EEBA7A80
waiting for block...
1
block produced
done
Import the following mnemonic into Keplr:
survey thank matrix joke trim more make gossip spread yellow unfold under cash beach harsh fire blush achieve oak swamp pluck clock rocket leg

The resulting address should be: agoric1xe269y3fhye8nrlduf826wgn499y6wmnv32tw5

make: Leaving directory '/workspace/contract'
✨  Done in 203.29s.
```

</details>

**Comment by migueldingli1997** [comment]
Something I'm suspecting (but have no direct evidence to back it up) is that the problems I'm facing are due to the container running slower than expected because of Mac M1 chip. Today I've installed a recently released Docker update that mentions better performance. I'll see if this has any impact on proposals passing or the sequence mismatch errors.

**Comment by dckc** [comment]
> Something I'm suspecting (but have no direct evidence to back it up) is that the problems I'm facing are due to the container running slower than expected because of Mac M1 chip.

That would be consistent with my "unless the proposal timed out?" hypothesis.

Lack of timestamps in some places is a pain. A lot of the output of `yarn start:contract` is messy. Here's hoping we address that:
 - #62
